sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -152,6 +152,11 @@ trait FileFormat {
     }
   }
 
+  /**
+   * Returns whether this format supports the given [[DataType]] in read/write path.
+   * By default all data types are supported.
+   */
+  def supportDataType(dataType: DataType, isReadPath: Boolean): Boolean = true
```

Comment:
who does not overwrite it?

Reply:
`HiveFileFormat`. Currently I don't know Hive well. If someone can override it for `HiveFileFormat`, please create a follow up PR.

Reply:
then why not remove this default implementation and create `HiveFileFormat#supportDataType` to return true?

Reply:
Then we also need to update `LibSVMFileFormat`, and several file format in unit test. I really prefer to have a default behavior here, as `FileFormat` can still work without the new method.

Reply:
makes sense